### PR TITLE
Upgrade nginx and add molecule support

### PR DESCRIPTION
**Pull request by gtirloni**
This PR is changing a few things:

 * Started to use official nginx repository package
 * Stopped using hardcoded URL to nginx package (pinned version instead of hardcoding URL)
 * Removed tasks that are unnecessary in nginx 1.12
 * Not considering Docker scenario any longer
 * Added molecule 2.x support (single scenario since this is a base role)
 * Fixed yamllint warnings/errors
 * Fixed copyright

Role was re-initialized for molecule 2.x using `molecule init role`.

### CODE REVIEW

File: molecule/default/.molecule/Vagrantfile [hunk 1]
```diff
@@ -0,0 +1,157 @@
+# Molecule managed
+
+require 'yaml'
+
+Vagrant.configure('2') do |config|
+  vagrant_config = YAML::load_file('/home/gtirloni/code/gtirloni-idrc/ansible-nginx-common/molecule/default/.molecule/vagrant.yml')
```
Comment (mrtyler) [inline molecule/default/.molecule/Vagrantfile]: This path doesn't seem right. Maybe this file should not be committed (and instead .gitignore'd)?

File: molecule/default/.molecule/ansible_inventory.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# Molecule managed
+
+---
+all:
+  hosts:
+    instance: &id001
+      ansible_host: 127.0.0.1
+      ansible_port: '2201'
+      ansible_private_key_file: /home/gtirloni/code/gtirloni-idrc/ansible-nginx-common/molecule/default/.molecule/.vagrant/machines/instance/virtualbox/private_key
```
Comment (mrtyler) [inline molecule/default/.molecule/ansible_inventory.yml]: Maybe all of .molecule should be un-committed/.gitignore'd?

File: molecule/default/.molecule/Vagrantfile (hunk 1 above)
Comment (gtirloni) [reply molecule/default/.molecule/Vagrantfile]: Good catch!

**Comment by mrtyler** [comment]
still lgtm :)

**Pull request merged**